experanto/datasets.py

```diff
@@ -0,0 +1,448 @@
+
+        screen:
+          sampling_rate: null
+          chunk_size: null
+          valid_condition:
+            tier: test
+            stim_type: stimulus.Frame
+          offset: 0
+          sample_stride: 4
+          include_blanks: false
+          transforms:
+            ToTensor:
+              _target_: torchvision.transforms.ToTensor
+            Normalize:
+              _target_: torchvision.transforms.Normalize
+              mean: 80.0
+              std: 60.0
+            Resize:
+              _target_: torchvision.transforms.Resize
+              size:
+              - 144
+              - 256
+            CenterCrop:
+              _target_: torchvision.transforms.CenterCrop
+              size: 144
+          interpolation: {}
+        responses:
+          sampling_rate: null
+          chunk_size: null
+          offset: 0.1
+          transforms:
+            standardize: true
+          interpolation:
+            interpolation_mode: nearest_neighbor
+        eye_tracker:
+          sampling_rate: null
+          chunk_size: null
+          offset: 0
+          transforms:
+            normalize: true
+          interpolation:
+            interpolation_mode: nearest_neighbor
+        treadmill:
+          sampling_rate: null
+          chunk_size: null
+          offset: 0
+          transforms:
+            normalize: true
+          interpolation:
+            interpolation_mode: nearest_neighbor
+        """
+        self.root_folder = Path(root_folder)
+        self.modality_config = instantiate(modality_config)
+        self.chunk_sizes, self.sampling_rates, self.chunk_s = {}, {}, {}
+        for device_name in self.modality_config.keys():
+            cfg = self.modality_config[device_name]
+            self.chunk_sizes[device_name] = global_chunk_size or cfg.chunk_size
+            self.sampling_rates[device_name] = global_sampling_rate or cfg.sampling_rate
+
+        self.sample_stride = self.modality_config.screen.sample_stride
+        self._experiment = Experiment(
+            root_folder,
+            modality_config,
+        )
+        self.device_names = self._experiment.device_names
+        self.start_time, self.end_time = self._experiment.get_valid_range("screen")
+        self._read_trials()
+        self.initialize_statistics()
+        self._screen_sample_times = np.arange(
+            self.start_time, self.end_time, 1.0 / self.sampling_rates["screen"]
+        )
+        # iterate over the valid condition in modality_config["screen"]["valid_condition"] to get the indices of self._screen_sample_times that meet all criteria
+        self._sample_in_meta_condition = self.get_sample_in_meta_condition()
+        self._full_valid_sample_times = self.get_full_valid_sample_times()
+        # the _valid_screen_times are the indices from which the starting points for the chunks will be taken
+        self._valid_screen_times = self._full_valid_sample_times[::self.sample_stride]
```

Comment:
This might be a bit dangerous in general scenario, as if the chunk and the sample_stride are big we might throw away too much data. 
I do not have a good solution suggestion, so a warning could help at least to make users aware.

Reply:
Yes agreed. The behavior of this should be made clearer. 

I'd argue that for our case, mostly these two cases are relevant:
- stride=1, this is for training, so that every sample can be a valid starting point
- stride = chunk_size, without randomization. This would mean that every frame gets sampled once, like in the validation and test set

Reply:
I added more comments here.